api/api_public_klay.go

```diff
@@ -70,3 +75,61 @@ func (s *PublicKlayAPI) Syncing() (interface{}, error) {
 		"knownStates":   hexutil.Uint64(progress.KnownStates),
 	}, nil
 }
+
+// EncodeAccountKey gets an account key of JSON format and returns RLP encoded bytes of the key.
+func (s *PublicKlayAPI) EncodeAccountKey(accKey accountkey.AccountKeyJSON) (hexutil.Bytes, error) {
+	buffer := new(bytes.Buffer)
+	if accKey.KeyType == nil {
+		return nil, errors.New("key type is not specified")
+	}
+	key, err := accountkey.NewAccountKey(*accKey.KeyType)
+	if err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(accKey.Key, key); err != nil {
+		return nil, err
+	}
+	// Invalidate zero values of threshold and weight to prevent users' mistake
+	// JSON unmarshalling sets zero for those values if they are not exist on JSON input
+	if err := checkAccountKeyZeroValues(key); err != nil {
+		return nil, err
+	}
+	accKeySerializer := accountkey.NewAccountKeySerializerWithAccountKey(key)
+	if err := accKeySerializer.EncodeRLP(buffer); err != nil {
+		return nil, errors.New("the key probably contains an invalid public key: " + err.Error())
+	}
+	return (hexutil.Bytes)(buffer.Bytes()), nil
+}
+
+// DecodeAccountKey gets an RLP encoded bytes of an account key and returns the decoded account key.
+func (s *PublicKlayAPI) DecodeAccountKey(encodedAccKey hexutil.Bytes) (*accountkey.AccountKeySerializer, error) {
+	dec := accountkey.NewAccountKeySerializer()
+	if err := rlp.DecodeBytes(encodedAccKey, &dec); err != nil {
+		return nil, err
+	}
+	return dec, nil
+}
+
+// checkAccountKeyZeroValues returns errors if the input account key contains zero values of threshold or weight.
+func checkAccountKeyZeroValues(key accountkey.AccountKey) error {
+	switch key.Type() {
+	case accountkey.AccountKeyTypeWeightedMultiSig:
+		multiSigKey, _ := key.(*accountkey.AccountKeyWeightedMultiSig)
+		if multiSigKey.Threshold == 0 {
+			return errors.New("invalid threshold of the multiSigKey")
+		}
+		for _, weightedKey := range multiSigKey.Keys {
+			if weightedKey.Weight == 0 {
+				return errors.New("invalid weight of the multiSigKey")
+			}
+		}
+	case accountkey.AccountKeyTypeRoleBased:
+		roleBasedKey, _ := key.(*accountkey.AccountKeyRoleBased)
+		for _, roleKey := range *roleBasedKey {
+			if err := checkAccountKeyZeroValues(roleKey); err != nil {
```

Comment:
Don't we need to check the call depth? What happens if there is a role-based key in a role-based key recursively?

Reply:
I limited the call depth to two.